### PR TITLE
Refactor auth tests to BDD, add edge-case tests, and harden JwtAuthFi…

### PR DESCRIPTION
**Pull request by AyoubGhouzali**
This PR refactors our authentication test suite to a Given–When–Then (BDD) style, adds edge-case scenarios (malformed/expired tokens, missing credentials, registration flows), and improves the JwtAuthFilter to catch and return 401 on invalid JWTs.

### CODE REVIEW

**Comment by kei077** [comment]
This branch contains outdated code. Though the merge would cause no conflicts. There are some files missing (jenkins file, docker compose files + I already deleted the static folder from the ressources directory )

please pull the code from the main branch before pushing the code again to this branch !

File: backend/src/test/java/com/cosmic/tracker/authTests/AuthRestControllerIntegrationTest.java [hunk 1]
```diff
@@ -4,34 +4,58 @@
...
+import java.security.Key;
+import java.util.Base64;
+import java.util.Date;
+
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
 
-@SpringBootTest
+@SpringBootTest(
```
Comment (kei077) [inline backend/src/test/java/com/cosmic/tracker/authTests/AuthRestControllerIntegrationTest.java:27]: Since we already have an application-test.properties with the same db properties you specified here you can just add a `@ActiveProfiles("test")` annotation and remove everything inside the `@SpringBootTest` annotation (but keep it just remove webEnvironment and properties specified inside) the test should still work just fine !

File: backend/src/test/java/com/cosmic/tracker/authTests/AuthRestControllerIntegrationTest.java
Comment (kei077) [inline backend/src/test/java/com/cosmic/tracker/authTests/AuthRestControllerIntegrationTest.java:1]: Everything looks good but I'm just wondering why you changed all the methods from public to package private ? Its fine to keep them that way bit is there a reason why we need our test methods as package private ?

File: backend/src/test/java/com/cosmic/tracker/authTests/AuthRestControllerIntegrationTest.java (hunk 1 above)
Comment (AyoubGhouzali) [reply backend/src/test/java/com/cosmic/tracker/authTests/AuthRestControllerIntegrationTest.java]: I tried it and it works properly.

File: backend/src/test/java/com/cosmic/tracker/authTests/AuthRestControllerIntegrationTest.java
Comment (AyoubGhouzali) [reply backend/src/test/java/com/cosmic/tracker/authTests/AuthRestControllerIntegrationTest.java:1]: Here is what I found:
"By declaring them package-private you:

Limit visibility to only the test package, which is a good “least privilege” practice.

Avoid clutter in your IDE’s auto-completion (only the methods you actually need to see are exposed).

Signal intention that these methods are strictly for testing, not part of any public API."

**Review by kei077 — APPROVED** [review]
Great job Ayoub ! Everything looks good ! 

Thank you for your contribution !

File: backend/src/test/java/com/cosmic/tracker/authTests/AuthRestControllerIntegrationTest.java
Comment (kei077) [reply backend/src/test/java/com/cosmic/tracker/authTests/AuthRestControllerIntegrationTest.java:1]: Sounds about right ! Already approved the PR just came back to reread, as I said it doesnt cause any issue but I guess what you found makes sense  !
